packages/venia-ui/lib/components/AccountChip/accountChip.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { shape, string } from 'prop-types';
+import { Loader, User as AccountIcon } from 'react-feather';
+
+import { useAccountChip } from '@magento/peregrine/lib/talons/AccountChip/useAccountChip';
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Icon from '../Icon';
+import defaultClasses from './accountChip.css';
+
+const AccountChip = props => {
+    const { fallbackText } = props;
+
+    const talonProps = useAccountChip({ fallbackText });
+    const { isLoadingUserName, welcomeMessage } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    const chipText = isLoadingUserName ? (
+        <Icon classes={{ icon: classes.loader }} src={Loader} />
```

Comment:
This feels weird and also strangely similar to [something I noticed in another PR](https://github.com/magento/pwa-studio/pull/2509#discussion_r453929810). I realize you created this component to be used in the header and in the left nav, so I think that influenced the direction. We may want to have a team discussion about when to use a loading message vs a spinner vs static text. In my opinion, showing the fallback text until the login operation completes would look better considering we already show a loading spinner for the content of the left drawer. For the header chip this might be OK.

[![Image from Gyazo](https://i.gyazo.com/fa2ca2ea555e14730d0d480b6cdc0add.gif)](https://gyazo.com/fa2ca2ea555e14730d0d480b6cdc0add)

Reply:
Resolved in https://github.com/magento/pwa-studio/pull/2550/commits/ef9e27a449816d1bbc59d013f17d3f8c0834ba56.